.github/workflows/tests.yml

```diff
@@ -1,7 +1,31 @@
 name: tests
-on: [push, pull_request]
+on: [push, pull_request, workflow_dispatch]
+
 jobs:
-  run_tests:
+  checkstyle:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+      - name: Cache Maven packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ matrix.job.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ matrix.job.os }}-m2
+
+      - name: Run checkstyle
+        run: mvn checkstyle:check
```

Comment:
Can we add fail fast here? If checkstyle fails, let's not even run the tests.

Reply:
I think they are skipped automatically by the use of .needs, but I will add it anyways to make sure

Reply:
I think they are skipped automatically by the use of .needs, but I will add it anyways to make sure